src/Runner.Worker/Container/ContainerHooks/ContainerHookManager.cs

```diff
@@ -0,0 +1,308 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using GitHub.DistributedTask.Pipelines.ContextData;
+using GitHub.DistributedTask.WebApi;
+using GitHub.Runner.Common;
+using GitHub.Runner.Common.Util;
+using GitHub.Runner.Sdk;
+using GitHub.Runner.Worker.Handlers;
+using GitHub.Services.WebApi;
+using Newtonsoft.Json.Linq;
+
+namespace GitHub.Runner.Worker.Container.ContainerHooks
+{
+    [ServiceLocator(Default = typeof(ContainerHookManager))]
+    public interface IContainerHookManager : IRunnerService
+    {
+        Task PrepareJobAsync(IExecutionContext context, List<ContainerInfo> containers);
+        Task CleanupJobAsync(IExecutionContext context, List<ContainerInfo> containers);
+        Task RunContainerStepAsync(IExecutionContext context, ContainerInfo container, string dockerFile);
+        Task RunScriptStepAsync(IExecutionContext context, ContainerInfo container, string arguments, string fileName, IDictionary<string, string> environment, string prependPath, string workingDirectory);
```

Comment:
```suggestion
        Task RunScriptStepAsync(IExecutionContext context, ContainerInfo container, string workingDirectory, string fileName, string arguments, IDictionary<string, string> environment, string prependPath);
```

rerange the args to match stepHost.executeasync